cibuildwheel/windows.py

```diff
@@ -115,6 +122,50 @@ def install_pypy(tmp: Path, arch: str, url: str) -> Path:
     return installation_path / "python.exe"
 
 
+def setup_setuptools_cross_compile(
+    python_configuration: PythonConfiguration,
+    python_libs_base: Path,
+    env: Dict[str, str],
+) -> None:
+    # We write to distutils_cfg for distutils-based builds because we know we
+    # currently don't have one (unless it's our own from a previous build)
+    for p in env["PATH"].split(os.pathsep):
+        distutils_cfg = Path(p) / "Lib/site-packages/setuptools/_distutils/distutils.cfg"
+        if distutils_cfg.parent.is_dir():
+            break
+    else:
+        log.warning("Did not find setuptools install to configure")
+        return
+
+    # Ensure our additional import libraries are made available, and explicitly
+    # set the platform name
+    map_plat = {"32": "win32", "64": "win-amd64", "ARM64": "win-arm64"}
+    plat_name = map_plat[python_configuration.arch]
+    # (Sorry, this file must be default/locale encoding.)
+    with distutils_cfg.open("w") as f:
+        print("[build]", file=f)
+        print("plat_name=", plat_name, file=f, sep="")
+        print("[build_ext]", file=f)
+        print("library_dirs=", python_libs_base, file=f, sep="")
+        print("plat_name=", plat_name, file=f, sep="")
+        print("[bdist_wheel]", file=f)
+        print("plat_name=", plat_name, file=f, sep="")
+
+    # setuptools builds require explicit override of PYD extension
+    # This is because it always gets the extension from the running
+    # interpreter, and has no logic to construct it. Currently, CPython's
+    # extensions follow our identifiers, but if they ever diverge in the
+    # future, we will need to store new data
+    env["SETUPTOOLS_EXT_SUFFIX"] = f".{python_configuration.identifier}.pyd"
+
+    # Cross-compilation requires fixes that only exist in setuptools's copy of
+    # distutils, so ensure that it is activated
+    # Since not all projects can handle the newer distutils, display a warning
+    # to help them figure out what may have gone wrong if this breaks for them
+    log.warning("Setting SETUPTOOLS_USE_DISTUTILS=local as it is required for cross-compilation")
```

Comment:
I think that a notice annotation might be more appropriate for this, that could be implemented in logger.py as a method like:

```python
    def notice(self, message: str) -> None:
        if self.fold_mode == "github":
            print(f"::notice::{message}\n", file=sys.stderr)
        else:
            c = self.colors
            print(f"{c.blue}Notice{c.end}: {message}\n", file=sys.stderr)
```